compiler/rustc_parse/src/parser/expr.rs

```diff
@@ -1750,7 +1750,24 @@ impl<'a> Parser<'a> {
     /// Parses an `if` expression (`if` token already eaten).
     fn parse_if_expr(&mut self, attrs: AttrVec) -> PResult<'a, P<Expr>> {
         let lo = self.prev_token.span;
-        let cond = self.parse_cond_expr()?;
+        let cond = self.parse_cond_expr()?.into_inner();
+
+        let cond = if let ExprKind::Paren(paren) = &cond.kind {
+            let peeled = paren.peel_parens();
+            if let ExprKind::Let(_, _) = peeled.kind {
+                // A user has written `if (let <pat> = <expr>)` (with some number of parens), we
+                // want to avoid confusing them with mentions of nightly features. If this logic is
+                // changed, you will also likely need to touch `unused::UnusedParens::check_expr`.
+                self.if_let_expr_with_parens(&cond, peeled);
+                cond.peel_parens_owned()
+            } else {
+                cond
+            }
+        } else {
+            cond
+        };
```

Comment:
Reassigning here would look better:

```suggestion
        let mut cond = self.parse_cond_expr()?.into_inner();

        if let (ExprKind::Paren(paren), false) = (&cond.kind, <feature_gate>) {
            let peeled = paren.peel_parens();
            if let ExprKind::Let(_, _) = peeled.kind {
                // A user has written `if (let <pat> = <expr>)` (with some number of parens), we
                // want to avoid confusing them with mentions of nightly features. If this logic is
                // changed, you will also likely need to touch `unused::UnusedParens::check_expr`.
                self.if_let_expr_with_parens(&cond, peeled);
                cond = cond.peel_parens_owned();
            }
        }
```

Reply:
This is just a preference, no? Why do you think it would be better?

Reply:
It is just a preference indeed. Personally it seems a bit easier to follow and tell what's going on: "in this _one_ chain of conditions, we change `cond`".

Reply:
> Personally it seems a bit easier to follow and tell what's going on: "in this one chain of conditions, we change cond".

Fair enough. I'll update.